python/res/fm/rms/rms_run.py

```diff
@@ -153,22 +153,22 @@ def exec_rms(self, exec_env):
             args += ["-threads", str(self.config.threads)]
 
         if exec_env:
-            env = os.environ.copy()
-            for key, value in exec_env.items():
-                if value is None:
-                    env.pop(key)
-                    continue
-
-                value = str(value).strip()
-                if len(value) == 0:
-                    env.pop(key)
-                    continue
-
-                env[key] = value
-
-            os.execve( self.config.executable , args, env )
+            os.execve(self.config.executable, args, exec_env)
         else:
-            os.execv(self.config.executable, args )
-
-
+            os.execv(self.config.executable, args)
+
+
+    @staticmethod
```

Comment:
This has nothing to do with the class per see, I suggest you move the implementation out of the class and put it above or below in this file... Also, remember to consider whether functions are part of the public API. If not, prefixing by `_` is not a bad idea ;)